GSOM.py

```diff
@@ -0,0 +1,103 @@
+from neuron import Neuron
+import numpy as np
+
+
+class GSOM:
+    def __init__(self, initial_map_size, parent_quantization_error, t1, t2, growing_metric, weights_vectors_dict=None):
+        self.__t1 = t1
+        self.__t2 = t2
+        self.__parent_quantization_error = parent_quantization_error
+        self.__map_size = initial_map_size
+        self.__growing_metric = growing_metric
+
+        # i'm not so sure...
+        self.neurons_map = np.asarray(
+            a=[[None for _ in range(initial_map_size[0])] for _ in range(initial_map_size[1])],
+            dtype=object
+        )
```

Comment:
questa neurons_map è da rivedere, secondo me è semplificabile

Reply:
anche secondo me. Solo non mi veniva in mente un modo più semplice per creare una struttura che potesse essere facilmente ingrandita (si deve aggiungere righe e colonne durante il train)